src/zapv2/__init__.py

```diff
@@ -146,7 +146,13 @@ def _request_api(self, url, query=None):
           # Add the apikey to get params for backwards compatibility
           if not query.get('apikey'):
             query['apikey'] = self.__apikey
-        return self.session.get(url, params=query, proxies=self.__proxies, verify=False)
+
+        response = self.session.get(url, params=query, proxies=self.__proxies, verify=False)
+
+        if (validate_status_code and response.status_code >= 300):
+            raise Exception("Invalid status code returned from zap, which indicate failure: " + str(response.status_code))
```

Comment:
- We could differentiate the errors, 400 (e.g. bad API request, invalid data) and 500 (e.g. issue in ZAP)?
 - Worth including the text of the response in the exception? That usually contains the cause of the error.

Reply:
I can differentiate the errors, is it normal to create specific exceptions in python? Or have you thought about different messages? I can also pass the status code. 
And yes, your correct, I'll add it.

Reply:
Yeah, I was thinking just on different messages.